digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/ProjectServiceSpec.scala

```diff
@@ -1159,4 +1159,136 @@ class ProjectServiceSpec extends FunSuite with Matchers with BeforeAndAfter {
       }
     }
   }
+  test("two projects with same road part") {
+    runWithRollback {
+      val error = intercept[RuntimeException] {
+        val rap1 = RoadAddressProject(0L, ProjectState.apply(1), "TestProject", "TestUser", DateTime.parse("1901-01-01"),
+          "TestUser", DateTime.parse("1901-01-01"), DateTime.now(), "Some additional info",
+          Seq(), None)
+        val rap2 = RoadAddressProject(0L, ProjectState.apply(1), "TestProject2", "TestUser", DateTime.parse("1901-01-01"),
+          "TestUser", DateTime.parse("1901-01-01"), DateTime.now(), "Some additional info",
+          Seq(), None)
+        val addr1 = List(ReservedRoadPart(Sequences.nextViitePrimaryKeySeqValue, 5, 207, Some(0L), Some(Continuous), Some(8L), None, None, None, None, true))
+        val project1 = projectService.createRoadLinkProject(rap1)
+        mockForProject(project1.id, RoadAddressDAO.fetchByRoadPart(5, 207).map(toProjectLink(project1)))
+        projectService.saveProject(project1.copy(reservedParts = addr1))
+        val project2 = projectService.createRoadLinkProject(rap2)
+        mockForProject(project2.id, RoadAddressDAO.fetchByRoadPart(5, 207).map(toProjectLink(project2)))
+        projectService.saveProject(project2.copy(reservedParts = addr1))
+      }
+      error.getMessage should be ("Can't reserve road part 5/207")
```

Comment:
error message should be :Tie <ROAD_NUMBER> osa <ROAD_PART_NUMBER> ei ole vapaana projektin alkupäivämääränä. Tieosoitteen alkupäivämäärä on sama kuin projektin alkupäivämäärä"

Reply:
Tie <ROAD_NUMBER> osa <ROAD_PART_NUMBER> ei ole vapaana projektin alkupäivämääränä. Tieosoite on jo varattuna toisessa projektissa <PROJECT_NAME>

Reply:
we seem to have case to already give this error message, but would be nice if test would also confirm this

Reply:
I agree.  Lets change it